src/app/view/Limits/hooks/useOAuthClientsLimitExceeded.ts

```diff
@@ -38,18 +39,28 @@ export const useOAuthClientsLimitExceeded = (
 
   useEffect(() => {
     const eventCallback = (href: string): void => {
-      if (client === null) {
-        log.error('Client is null, should not happen')
-        return
+      const doAsync = async (): Promise<void> => {
```

Comment:
looks complex for hook logic, I think we could extract that logic outside of react, especially since it doesn't seem closely coupled to react context values

Reply:
I moved it in those commits:
- https://github.com/cozy/cozy-flagship-app/pull/1143/commits/1ef7dcba7fdaa406c53fa6824629c547ab40c72f
- https://github.com/cozy/cozy-flagship-app/pull/1143/commits/790f3a4d4defea638914e782b5905941310a0549